Makefile

```diff
@@ -116,30 +117,30 @@ go/deps:
 	$(GO) mod tidy
 
 define go_build_pyroscope
-	GOOS=$(GOOS) GOARCH=$(GOARCH) CGO_ENABLED=0 $(GO) build -tags "netgo $(EMBEDASSETS)" -ldflags "-extldflags \"-static\" $(1)" ./cmd/pyroscope
+	GOOS=$(GOOS) GOARCH=$(GOARCH) CGO_ENABLED=0 $(GO) build -tags "netgo $(EMBEDASSETS)" -ldflags "-extldflags \"-static\" $(1)" -gcflags=$(2) ./cmd/pyroscope
 endef
 
 define go_build_profilecli
-	GOOS=$(GOOS) GOARCH=$(GOARCH) CGO_ENABLED=0 $(GO) build -ldflags "-extldflags \"-static\" $(1)" ./cmd/profilecli
+	GOOS=$(GOOS) GOARCH=$(GOARCH) CGO_ENABLED=0 $(GO) build -ldflags "-extldflags \"-static\" $(1)" -gcflags=$(2) ./cmd/profilecli
 endef
 
 .PHONY: go/bin-debug
 go/bin-debug:
-	$(call go_build_pyroscope,$(GO_LDFLAGS))
-	$(call go_build_profilecli,$(GO_LDFLAGS))
+	$(call go_build_pyroscope,$(GO_LDFLAGS),$(GO_GCFLAGS))
+	$(call go_build_profilecli,$(GO_LDFLAGS),$(GO_GCFLAGS))
 
 .PHONY: go/bin
 go/bin:
-	$(call go_build_pyroscope,-s -w $(GO_LDFLAGS))
-	$(call go_build_profilecli,-s -w $(GO_LDFLAGS))
+	$(call go_build_pyroscope,-s -w $(GO_LDFLAGS),)
+	$(call go_build_profilecli,-s -w $(GO_LDFLAGS),)
 
 .PHONY: go/bin-pyroscope-debug
 go/bin-pyroscope-debug:
-	$(call go_build_pyroscope,$(GO_LDFLAGS))
+	$(call go_build_pyroscope,$(GO_LDFLAGS),$(GO_GCFLAGS))
 
 .PHONY: go/bin-profilecli-debug
```

Comment:
nitpick and unrelated to this PR, this should probably match the make target (`go/bin-profilecli`)

Reply:
Fixed in https://github.com/grafana/pyroscope/pull/3517/commits/8d1035ab702ad26a754b45863a736a00456973e8!